### PR TITLE
Fe profile

### PR DESCRIPTION
**Pull request by vycdev**

### CODE REVIEW

File: packages/api/db/migrations/20200712133735_create_totaltests_column.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Knex from "knex";
+
+export const up = async (knex: Knex) => {
+    return knex.schema.table("users", table => {
```
Comment (BrandonHowe) [inline packages/api/db/migrations/20200712133735_create_totaltests_column.ts:4]: You can always just leave out return and the braces, see how I do it

File: packages/api/src/modules/users/actions/updateDescription.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import User from "../types/User";
+import knex from "../../../../db/knex";
+
+export default async <T extends keyof Omit<User, "password">>(
+    property: T,
+    value: User[T],
```
Comment (BrandonHowe) [inline packages/api/src/modules/users/actions/updateDescription.ts:6]: Seems like you always reuse the findUser code, why do you think the function is there? Just make it accept an ID

File: packages/api/src/modules/users/router.ts [hunk 3]
```diff
@@ -90,29 +91,39 @@ router.get("/userPBs/:id", async (ctx, next) => {
...
+// });
+
+router.get("/userData/:id", async (ctx, next) => {
+    const { id } = ctx.params;
+
+    const data = await getUserData("id", id);
 
     ctx.status = 200;
-    ctx.body = achievements;
+    ctx.body = { data };
```
Comment (BrandonHowe) [inline packages/api/src/modules/users/router.ts]: Why is this an object with 1 property?

File: packages/api/src/modules/users/router.ts [hunk 4]
```diff
@@ -133,4 +144,21 @@ router.post(
...
+    "/updatedescription",
+    requireAuthenticated(),
+    validateSchema(UpdateCountry, "body"),
+    async (ctx, next) => {
+        const { description } = ctx.request.body;
+        const { user } = ctx.session!;
+
+        await updateDescription("id", user, description);
+
+        ctx.status = 201;
```
Comment (BrandonHowe) [inline packages/api/src/modules/users/router.ts]:
```suggestion
        ctx.status = 200;
```

201 is for when you create something. Use 200.

File: packages/api/src/modules/users/router.ts [hunk 5]
```diff
@@ -133,4 +144,21 @@ router.post(
     }
 );
 
+router.post(
```
Comment (BrandonHowe) [inline packages/api/src/modules/users/router.ts]:
```suggestion
router.patch(
```

Use patch, not post, you are updating data.

File: packages/api/src/modules/users/router.ts [hunk 6]
```diff
@@ -133,4 +144,21 @@ router.post(
...
+    "/updatedescription",
```
Comment (BrandonHowe) [inline packages/api/src/modules/users/router.ts]:
```suggestion
    "/updateDescription",
```

camelCase please.

File: packages/api/src/modules/users/router.ts [hunk 7]
```diff
@@ -90,29 +91,39 @@ router.get("/userPBs/:id", async (ctx, next) => {
     await next();
 });
 
+// router.get("/userCountry/:id", async (ctx, next) => {
```
Comment (BrandonHowe) [inline packages/api/src/modules/users/router.ts]: Get rid of ghost code, please.

**Review by BrandonHowe — CHANGES_REQUESTED** [review]
Mostly good, you just need to change some stuff.

File: packages/api/src/modules/users/router.ts (hunk 6 above)
Comment (vycdev) [reply packages/api/src/modules/users/router.ts]: well its a route doesnt matter thatt much but I fixed it anyway 😄

File: packages/api/src/modules/users/router.ts (hunk 7 above)
Comment (vycdev) [reply packages/api/src/modules/users/router.ts]: Alright got rid of it

File: packages/api/db/migrations/20200712133735_create_totaltests_column.ts (hunk 1 above)
Comment (vycdev) [reply packages/api/db/migrations/20200712133735_create_totaltests_column.ts:4]: Well you don't always do it.

**Review by BrandonHowe — APPROVED** [review]

**Pull request merged**